### ISSUE TITLE
[Suggestion]Net Splitter Adapter

### ISSUE THREAD

**Issue by Player505** [MightyPirates/OpenComputers#2531]
I'd like it if I was able to have a net splitter next to an adapter block open up commands in the lua interpreter to specify which sides of the net splitter to pass a signal through (also the side that the net splitter and adapter block touch not allowed to pass a signal under any circumstances because that might cause problems)

**Comment by payonel** [comment]
As with any ticket that I set "open for adoption", an interested contributor should discuss the implementation details with us to make sure we're on the same page. For example, this cannot be signal specific. This cannot change the networking logic of how component discover and connect (specifically, leave the network/Network.scala unchanged). It would have to be localized to one block that can decide for itself which blocks are neighbors.

**Comment by Player505** [comment]
Can you please explain what you mean by signal specific?

**Comment by payonel** [comment]
reading the original ask more carefully, I now believe I misread - so please disregard my comment about signal specific filtering.
Sounds like this splitter would have the ability to act as an intersection for 6 networks. The 6 networks would be logically separate, but modem_messages could be repeated from one network to another.
That would require an network object that acts like 6 objects, depending on the side attached.

**Comment by Player505** [comment]
Sorry but I think I wasn't clear enough in my first explanation. [Here](https://imgur.com/a/vR0jx) are some images to help my explanation. The net splitter as it stands allows for components to be connected through them as shown in the second image. 

What I am trying to suggest is that the adapter block will let the computer at the back manipulate the NBT data of the net splitter (I use NBT data for the interest of this example even though I'm not sure how the information for which sides of the net splitter are active and which sides are not is stored) to control which sides allow a signal to pass through and the components to see each other and which side do not allow this. The command could be as follows,
```
component.netsplitter.active(side:number, activate:boolean): boolean
```

**Comment by Vexatos** [comment]
This would just need to be implemented as a trivial driver replicating the functionality a wrench has on a net splitter. It sounds extremely useful for automating network routes. You would have to take care of keeping the side the adapter is on closed yourself since forcing that when you attach the adapter probably wouldn't be great.

**Comment by Vexatos** [comment]
One thing you could do is disallow _enabling_ the side the adapter is on through the driver to prevent issues. Forcing the connection to be and stay off would be tricky to do though.

**Comment by Player505** [comment]
I was thinking disallowing the side that is touching the adaptor block to be enabled

**Comment by payonel** [comment]
we really like this suggestion, but, it's going to be a normal component that loads in the network and not through the adapter. If you close the side and thus sever your network connection to it, you will also lose the component. Perhaps not as magical as you were hoping, but you still get the net spliter component.

1. the adapter really shouldn't be used for intra-mod features, it is a cross mod block and after consultation I have decided it isn't right to add this to the adapter block. though -- that was a rather easy addition
2. after testing a lot of mechanics to allow the six sides to provide a component (whether or not there was a locked side), but having it be the same component...without also connecting the networks unless the splitter face was open -- just isn't possible under our current network design. The node comparisons are based on node guid, and the component add method has the same check. We simply cannot have a single component shared independently across 6 faces and not share their node networks, without building some special comparison check just for this one unusual feature. I investigated using a type of component inventory, proxy nodes, dynamic sided nodes, etc -- to make this work is a hack, and not how our networks are designed. Perhaps our networks should be more flexible, but that is a refactor and a major change, and not the right time just for this feature.
